lib/wibox/widget/textbox.lua

```diff
@@ -373,6 +373,26 @@ function textbox.mt.__call(_, ...)
     return new(...)
 end
 
+--- Get geometry of text label, as if textbox would be created for it on the screen.
+--
+-- @tparam string text The text content, pango markup supported.
+-- @tparam[opt=nil] integer|screen s The screen on which the textbox would be displayed.
+-- @tparam[opt=beautiful.font] string font The font description as string.
+-- @treturn table Geometry (width, height) hashtable.
+function textbox.get_markup_geometry(text, s, font)
+    font = font or beautiful.font
+    local pctx = PangoCairo.font_map_get_default():create_context()
+    local playout = Pango.Layout.new(pctx)
+    playout:set_font_description(beautiful.get_font(font))
+    local dpi_scale = beautiful.xresources.get_dpi(s)
+    pctx:set_resolution(dpi_scale)
+    playout:context_changed()
+    local attr, parsed = Pango.parse_markup(text, -1, 0)
+    playout.attributes, playout.text = attr, parsed
+    local _, logical = playout:get_pixel_extents()
+    return logical
+end
```

Comment:
Isn't the above equivalent to the following, already existing functionality?
```lua
local t = wibox.widget.textbox(text)
t:set_font(font)
local w, h = t:get_preferred_size(s)
return { width = w, height = h }
```
Why a new function for this?

Reply:
it is, but @Elv13 prefers it like that: https://github.com/awesomeWM/awesome/pull/3160#discussion_r482705037

Reply:
Hm, okay... @Elv13 why? This adds new public API for something that is already possible and creating a throw-away widget should be so cheap that it doesn't make much difference that some more tables are created.

Reply:
At least there should be more comments in this new function to explain the use case and how it works.

Reply:
@keryell what exactly would you add there?

Reply:
like what?

Reply:
ping @Elv13

Reply:
@Elv13 , bump

Reply:
If psychon prefers not adding a new API, then I backtrack. Let's not add a new API and keep using this hack in 9999 places (but most of them are test code, so they don't count)

Reply:
so, @psychon ?

Reply:
btw, while fixing some stuff in menubar i also found what it have similar functions already:
https://github.com/awesomeWM/awesome/blob/master/lib/menubar/utils.lua#L413-L428

so mb it would be better to merge this as is, and next in a separate PR to move and deprecate the one from menubar.utils

the important difference, what the new function from this PR is taking font into consideration (while the one in menubar is actually broken, as it will show wrong value if using non-default font or even font size in menubar)

Reply:
If you think this is useful, then go ahead and add a new API for this.

Reply:
However... would it make sense to implement this new function via my suggestion above? That would make the code a lot shorter.
```lua
local t = wibox.widget.textbox(text)
t:set_font(font)
local w, h = t:get_preferred_size(s)
return { width = w, height = h }
```

Reply:
i personally don't mind this implementation, the concern about it was what it creates the whole widget object without a reason